composeApp/src/commonMain/kotlin/org/ooni/probe/domain/ObserveAndConfigureAutoUpdate.kt

```diff
@@ -0,0 +1,28 @@
+package org.ooni.probe.domain
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import org.ooni.probe.data.models.InstalledTestDescriptorModel
+import kotlin.coroutines.CoroutineContext
+
+class ObserveAndConfigureAutoUpdate(
+    private val backgroundContext: CoroutineContext,
+    private val listAllInstalledTestDescriptors: () -> Flow<List<InstalledTestDescriptorModel>>,
+    private val configureDescriptorAutoUpdate: suspend () -> Boolean,
+    private val cancelDescriptorAutoUpdate: suspend () -> Boolean,
+    private val startDescriptorsUpdate: suspend (List<InstalledTestDescriptorModel>?) -> Unit,
+) {
+    operator fun invoke() =
+        listAllInstalledTestDescriptors()
```

Comment:
Just a small thing. This will trigger a new value on every change to descriptors. If you make it like this, it will trigger only when we need to:
```
listAllInstalledTestDescriptors()
   .map { it.isNotEmpty() }
   .distinctUntilChanged()
   .onEach { enabled -> ...
```